src/main/java/org/spongepowered/api/event/entity/living/player/PlayerResourcePackStatusEvent.java

```diff
@@ -0,0 +1,63 @@
+package org.spongepowered.api.event.entity.living.player;
+
+import org.spongepowered.api.resourcepack.ResourcePack;
+
+import com.google.common.base.Optional;
+
+public interface PlayerResourcePackStatusEvent extends PlayerEvent {
+
+    /**
+     * Gets the pack that this status corresponds to.
+     * 
+     * @return The pack that this status corresponds to.
+     */
+    ResourcePack getPack();
+
+    /**
+     * Gets the status of the sent pack.
+     * 
+     * @return The status of the sent pack.
+     */
+    ResourcePackStatus getStatus();
+
+    public enum ResourcePackStatus {
+        /**
+         * The client is attempting to download the pack.
+         */
+        ACCEPTED(Optional.<Boolean> absent()),
```

Comment:
IMHO you should use plain input here (here null) and then wrap them in the constructor. That way its more readable and secure because you cannot accidentially create an intance with null as successValue.